model/src/main/proto/exploration_checkpoint.proto

```diff
@@ -27,6 +27,12 @@ message ExplorationCheckpoint {
   // Stores answers to the latest state in the deck.
   repeated AnswerAndResponse pending_user_answers = 3;
 
+  // Stores whether the current selected state should show the 'learn again' button or not.
+  bool show_learn_again_button = 11;
```

Comment:
What's the repercussion for not doing this? That is, if we don't implement the checkpoint changes what becomes the user experience?

I'm hesitant to build changes into checkpoints because the full version of this feature will change them yet again, and we'd need to consider introducing a migration pathway which might be quite complex. Instead, I think it's fine to have a suboptimal checkpointing experience for this edge case until at least the full solution is in place.